pkg/buildpack/package.go

```diff
@@ -155,6 +204,27 @@ func extractExtensions(pkg Package) (mainExt BuildModule, err error) {
 	return mainExt, nil
 }
 
+type void struct{}
+
+var member void
+
+func flattenedBuildpacksDiffIDs(layers dist.ModuleLayers) map[string]struct{} {
+	var seenDiffID = make(map[string]struct{})
+	var flattenedDiffIDs = make(map[string]struct{})
```

Comment:
Why do we need two maps?

Reply:
I added a new method `ContainsFlattenedModules() bool` to the BuildModule interface. 

Let's suppose we have a. flattened buildpack with buildpacks [A,B,C] inside of it.

- seenDiffID: tracks the second time you find the same DiffID, this is useful to return the empty reader. In the example, it will tell us to return an empty reader for B and C for example
- flattenedDiffIDs: tracks all the build modules associated with a particular DiffID, basically to being able to return *true* with the new method added. In the example, it will tell me [A,B,C] are flattened. 

Basically is for keep track of all the buildpacks associated with a DiffID.

Reply:
Oic... what about `var seenDiffID = make(map[string]bool` where `bool` is "is flattened" (or not)?

Reply:
mmm, lets see. 
`var seenDiffID = make(map[string]bool)` the logic will be something like:

```golang
if _, ok := seenDiffID[bpInfo.LayerDiffID]; ok {
  // second time we see the same DiffID
  seenDiffID[bpInfo.LayerDiffID] = true
} else {
  // first time we see the DiffID, we don't know if it is flattened
  seenDiffID[bpInfo.LayerDiffID] = false
}

// some where out of the loop we will need to iterate over seenDiffID to just get those keys that are true (flattened) 
// or returned the map[string]bool and change the logic to check the key and the value right?

```